toolkit/ar/src/main/java/com/arcgismaps/toolkit/ar/WorldScaleSceneView.kt

```diff
@@ -244,14 +259,46 @@ public fun WorldScaleSceneView(
             // onResetOriginCamera is only called in WorldTracking mode. We need to reset the AR
             // session in WorldTracking Mode when the origin camera is reset, because AR tracking
             // will need to start from scratch.
-            arSessionWrapper.resetSession(worldScaleTrackingMode is WorldScaleTrackingMode.Geospatial)
+            arSessionWrapper.resetSession(
+                worldScaleTrackingMode is WorldScaleTrackingMode.Geospatial,
+                planeFindingMode
+            )
         }
     )
 
+    var lastSingleTapConfirmedEvent: SingleTapConfirmedEvent? by remember { mutableStateOf(null) }
+
     Box(modifier = modifier) {
         ArCameraFeed(
             session = arSessionWrapper,
             onFrame = { frame, displayRotation, session ->
+                // if the user-provided lambda callback is not null
+                onSingleTapConfirmed?.let { userSingleTapConfirmedCallback ->
+                    // if the last single tap confirmed event from the scene view is not null
+                    lastSingleTapConfirmedEvent?.let { singleTapConfirmedEvent ->
+                        val hitResult = frame.hitTest(
+                            singleTapConfirmedEvent.screenCoordinate.x.toFloat(),
+                            singleTapConfirmedEvent.screenCoordinate.y.toFloat()
+                        )
+                        val hit = hitResult.firstOrNull {
+                            when (it.trackable) {
+                                is Plane, is Point -> true
+                                else -> false
+                            }
+                        }
+                        if (hit == null) {
+                            userSingleTapConfirmedCallback(singleTapConfirmedEvent)
+                        }
+                        else {
+                            // hit is not null, so we can get the map point
+                            // first get the point relative to camera
+                            val mapPoint =
+                                worldScaleCameraController.getPointFromPose(hit.hitPose, session)
+                            userSingleTapConfirmedCallback(singleTapConfirmedEvent.copy(mapPoint = mapPoint))
+                        }
```

Comment:
formatting- move `else` one line up, just after the closing bracket of the `if` statement:
```suggestion
                        if (hit == null) {
                            userSingleTapConfirmedCallback(singleTapConfirmedEvent)
                        } else {
                            // hit is not null, so we can get the map point
                            // first get the point relative to camera
                            val mapPoint =
                                worldScaleCameraController.getPointFromPose(hit.hitPose, session)
                            userSingleTapConfirmedCallback(singleTapConfirmedEvent.copy(mapPoint = mapPoint))
                        }
```